src/main/java/jmgomez/policyapi/service/PolicyService.java

```diff
@@ -0,0 +1,48 @@
+package jmgomez.policyapi.service;
+
+import jmgomez.policyapi.feignClient.PolicyClient;
+import jmgomez.policyapi.mapper.AccidentMapper;
+import jmgomez.policyapi.mapper.PolicyMapper;
+import jmgomez.policyapi.model.dto.AccidentDto;
+import jmgomez.policyapi.model.dto.PolicyDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PolicyService implements IPolicyService {
+
+    @Autowired
+    private PolicyClient policyClient;
+
+    @Autowired
+    private PolicyMapper policyMapper;
+
+    @Autowired
+    private AccidentMapper accidentMapper;
+
+    @Override
+    public List<PolicyDto> getPolicies(String userId) {
+        return policyMapper.toListDto(policyClient.getPolicies(userId));
+    }
+
+    @Override
+    @Cacheable("policies")
+    public PolicyDto getPolicyByIDs(String policyId)  {
+        return policyMapper.toDto(policyClient.getPolicyByIDs(policyId));
+    }
+
+    @Override
+    public List<AccidentDto> getAccidents(String policyId) {
+        return accidentMapper.toListDto(policyClient.getAccidents(policyId));
+    }
+
+    @Override
+    @CachePut("accidents")
```

Comment:
Why CachePut here? It doesn't work like Cacheable

Reply:
I was testing the different cache types and at first sight I didn't appreciate the difference. I know that the CachePut is used when you have saved cache and you don't want to save more because it could decrease the effectivity of the application.

Reply:
Mmmmm... nope. CachePut is used to update the cache with the result in each execution. "Write only", easily explained. So next execution won't use the cache... I'd probably use CachePut in a write operation because after an update in the data you probably want to update a related cache and probably makes no sense to skip execution and return the data from cache. But we are talking about read operations. Makes no sense. Please, if you don't understand what I'm meaning, ask me in teams and we will go deep down.